sklearn/metrics/cluster/_dbcv_linkage.pyx

```diff
@@ -0,0 +1,42 @@
+import numpy as np
+cimport numpy as np
+
+
+cpdef np.ndarray[np.double_t, ndim=2] mst_linkage_core(
+                               np.ndarray[np.double_t,
+                                          ndim=2] distance_matrix):
```

Comment:
Please, can you reuse the existing implementation?

https://github.com/scikit-learn/scikit-learn/blob/981488412b7fcf41a3e1247e0efc5ca69a38e376/sklearn/cluster/_hierarchical_fast.pyx#L428-L430

Reply:
Alright, a naive approach for the modification is out now. Sorry it took so long for this simple change. Spent some time to understand both the linkage version of sklearn as well as HDBSCAN's. The sklearn version sure is more readable ^^ 

Unsure about the modification I made to the function signature (had to remove the types). I guess we'll see if it breaks anything once the CI has run

Reply:
As suspected, removing the typing in `mst_linkage_core`'s signature causes a problem when we try to pass the addresses needed for the `DistanceMetric64` calculations in the non-precomputed case: `Cannot take address of Python object`. I'm unsure what the best approach for a signature for the function would be. I tried to adhere more or less to what I'd consider the sklearn-idiomatic approach, which is that a single metric param can either be a callable or a string. Since we're doing Cython here though, I'm unsure. Maybe you @jjerphan have a suggestion for me?

Reply:
I guess I could try adding a new, boolean "precomputed" kwarg, set to `False` by default. If set to `True`, the caller can then simply pass `None` instead of an actual `DistanceMetric64` instance?

Reply:
Ok, I think I've got the typing working now. Also switched the call in the DBCV internals as well as cleaning up the meson cfg and deleting the extra linkage implementation that I brought along. However, all my builds after `buildId=64414` run into failing tests due to a bunch of `BrokenProcessPool` errors. Looking at them myself, I'm failing to realize how this could possibly be related to my changes. But I guess it must be my fault?

Another question that remains is why we are transposing (`::1`) the first argument of `mst_linkage_core` ... I am not really up to speed when it comes to typed memviews in Cython.

Reply:
Alright, I worked it out. The previous test failures were caused by importing errors, specifically due to a subpackage init cycle.

To circumvent that, I ultimately ended up having to use a [function-internal import statement](https://github.com/luis261/scikit-learn/blob/add-dbcv-metric/sklearn/metrics/cluster/_dbcv_internals.py#L130); granted, that's more efficient but arguably also a code smell indicative of a deeper, underlying problem. I guess there is an argument to be made that `DBCV` is actually more `cluster` than it is `metrics.cluster`, at least from an implementation perspective. Other than completely moving it to a somewhat unintuitive location though or completely rearchitecting the way the affected modules are distributed in their respective packages, I don't see an alternative to an import at function-runtime.